### PR TITLE
[Merged by Bors] - feat: port Data.Multiset.Nodup

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

- [x] depends on:  #1537
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Multiset/Nodup.lean [hunk 1]
```diff
@@ -15,7 +23,255 @@ open Function List
 
 variable {α β γ : Type _} {r : α → α → Prop} {s t : Multiset α} {a : α}
 
+-- nodup
 /-- `nodup s` means that `s` has no duplicates, i.e. the multiplicity of
```
Comment (jcommelin) [inline Mathlib/Data/Multiset/Nodup.lean:27]:
```suggestion
/-- `Nodup s` means that `s` has no duplicates, i.e. the multiplicity of
```

File: Mathlib/Data/Multiset/Nodup.lean [hunk 2]
```diff
@@ -2,8 +2,16 @@
...
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
 -/
-import Mathlib.Data.Multiset.Basic
+import Mathlib.Data.List.Nodup
+import Mathlib.Data.Multiset.Bind
+import Mathlib.Data.Multiset.Range
+
 /-!
 # The `nodup` predicate for multisets without duplicate elements.
```
Comment (jcommelin) [inline Mathlib/Data/Multiset/Nodup.lean:16]:
```suggestion
# The `Nodup` predicate for multisets without duplicate elements.
```

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1537~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**